addons/web/static/src/views/helpers/view_compiler.js

```diff
@@ -744,101 +744,48 @@ export class ViewCompiler {
-        }
-        notebook.classList.add("o_notebook");
-
-        const activePage = createElement("t");
-        activePage.setAttribute("t-set", noteBookId);
-        activePage.setAttribute(
-            "t-value",
-            `state.${noteBookId} or getActivePage(record,${JSON.stringify(invisibleDomains)})`
-        );
-        append(notebook, activePage);
-
-        const headers = createElement("div");
-        headers.className = "o_notebook_headers";
-
-        append(headers, headersList);
-        append(notebook, headers);
-        append(notebook, contents);
 
-        return notebook;
-    }
-
-    /**
-     * @param {Element} el
-     * @param {Record<string, any>} params
-     * @returns {{
-     *  pageId: string,
-     *  header: Element,
-     *  content: Element,
-     *  invisible: boolean | boolean[],
-     * }}
-     */
-    compilePage(el, params) {
-        const invisible = getModifier(el, "invisible");
-        if (typeof invisible === "boolean" && invisible) {
-            return;
-        }
+            const pageSlot = createElement("t");
+            append(noteBook, pageSlot);
 
-        const pageId = `page_${this.id++}`;
-        const header = createElement("li");
-        header.className = "nav-item";
-
-        const headerLink = createElement("a");
-        headerLink.className = "nav-link";
-        headerLink.setAttribute("t-on-click.prevent", `()=>state.${params.noteBookId}='${pageId}'`);
-        headerLink.setAttribute("href", "#");
-        headerLink.setAttribute("role", "tab");
-        headerLink.setAttribute(
-            "t-attf-class",
-            `{{${params.noteBookId}==='${pageId}'?'active':''}}`
-        );
-        headerLink.textContent = el.getAttribute("string") || el.getAttribute("name");
-        append(header, headerLink);
+            const pageId = `page_${this.id++}`;
+            const pageTitle = `"${child.getAttribute("string") || child.getAttribute("name")}"`; // LPE Fixme : escape chars (quotes etc...)
+            pageSlot.setAttribute("t-set-slot", pageId);
+            pageSlot.setAttribute("title", pageTitle);
 
-        const content = createElement("div");
-        content.setAttribute("t-if", `${params.noteBookId}==='${pageId}'`);
-        content.className = "tab-pane active";
+            let isVisible;
+            if (invisible === false) {
+                isVisible = "true";
+            } else {
+                isVisible = `!evalDomain(record,${JSON.stringify(invisible)})`;
+            }
+            pageSlot.setAttribute("isVisible", isVisible);
 
-        for (const child of el.childNodes) {
-            append(content, this.compileNode(child, params));
+            for (const contents of child.children) {
+                append(pageSlot, this.compileNode(contents, params));
+            }
+            // In owl 2 slots that have no content are not passed to the Component,
+            // so we force dummy content on the slot.
```

Comment:
Might be a bug?

Reply:
Why do we insert a Notebook component in that case?

Reply:
I suppose we may want a tab even if that tab is empty?

Reply:
There is a test case that asserts that if the page is visible (but empty) the tab of that page is rendered along with the notebook (Views > FormView: invisible elements are properly hidden).

I suspect this test only uses a side effect but doesn't really encode a spec

Reply:
I still think it should be supported in owl. We have the freedom to omit the page in the view_compiler if it's empty if we want to, but the presence of a slot is information that slot-using components may want to rely on even if the slot is empty (eg if you want to override the slot's default content with nothing)